src/sessionsConverter.js

```diff
@@ -0,0 +1,114 @@
+// module pattern
+var sessionsConverter = (function(){
+  // variable to keep track of different labels
+  var labels = [];
+  // labels set for checking if label is unique
+  var labelSet = {};
+
+  function init(){
+    labels = [];
+    labelSet = {};
+  }
+
+  function getConvertedSessions(sessions){
+    var convertedSessions = [];
+    sessions.forEach(function(session){
+      convertedSessions.push(getConvertedSession(session));
+    });
+    return convertedSessions;
+  }
+
+  function getConvertedSession(session){
+    var now = moment(); // current time
+
+    var outcomes = session.outcomes;
+    // data set item for session
+    var convertedSession = {};
+
+    var momentTimestamp = moment(session.timestamp);
+    var humanisedTimestamp = moment.duration(momentTimestamp.diff(now)).humanize(true);
+
+    convertedSession.label = humanisedTimestamp;
+
+    convertedSession.valuesMap = getValuesMapsFromOutcomes(outcomes);
+
+    // extract data and notes arrays
+    var extractedDataAndNotes = getExtractedDataAndTooltipNotes(convertedSession.valuesMap);
+    convertedSession.data = extractedDataAndNotes.data;
+    convertedSession.notes = extractedDataAndNotes.notes;
+
+
+    var color = Chart.helpers.color;
+    //assign random colour to chart
+    var colour = randomColor({
+      format: 'rgba'
+    });
+    convertedSession.backgroundColor = color(colour).alpha(0.2).rgbString();
+    convertedSession.borderColor = colour;
+    convertedSession.pointBackgroundColor = colour;
+    
+    return convertedSession;
+  }
+
+  function getValuesMapsFromOutcomes(outcomes){
+    var dataMap = {};
+    var noteMap = {};
+    // add each outcome to a set
+    outcomes.forEach(function(outcome){
+      var lowerCaseLabel = outcome.outcome.toLowerCase();
+      updateLabelSet(outcome.outcome);
+      dataMap[lowerCaseLabel] = outcome.value;
+      noteMap[lowerCaseLabel] = outcome.notes;
+    });
+    return {
+      data : dataMap,
+      notes: noteMap
+    }
+  }
+
+  // check if our set has the value or not
+  function updateLabelSet(potentialLabel){
+    var lowerCaseLabel = potentialLabel.toLowerCase();
+    if(!labelSet.hasOwnProperty(lowerCaseLabel)){
+      labels.push(potentialLabel);
+      // now value is in our set
+      labelSet[lowerCaseLabel] = true;
```

Comment:
instead of setting both `labels` and `labelSet`, you could completely remove `labels` and use just the dictionary. This would remove the `if` check here, and in the `getExtractedDataAndTooltipNotes` function you can iterate over `Object.keys(labelSet)` instead of `labels`